util/llpcElfWriter.cpp

```diff
@@ -955,6 +966,118 @@ void ElfWriter<Elf>::MergeElfBinary(
     WriteToBuffer(pPipelineElf);
 }
 
+template<class Elf>
+void ElfWriter<Elf>::Reinitialize() {
```

Comment:
why do we need this function? I think we can always create new instance when need.

Reply:
The `reinitialize` is private.  From the perspective if the class, it does not know if you uses the same instance to read from a buffer, and then link.  In this case, there could be old data from the buffer that was read hanging around that should not be in the result of the link.  When I go through the linking, I want to be able to make assumptions about the state of the sections, which is why I clear it.

I could have inlined this into `LinkRelocatableElf`, but it clutters that function needlessly.  This could also be used by other functions in the future if needed.